src/command_executor.cc

```diff
@@ -926,6 +927,31 @@ sql_callback(exec_context& ec, sqlite3_stmt* stmt)
             dls.push_header(colname, type, graphable);
             if (graphable) {
                 auto attrs = vc.attrs_for_ident(colname);
+                dls.dls_headers[lpc].hm_colorstr = colname;
+                for (
+                    std::string suf = " ";
+                    std::any_of(
+                        dls.fg_colors_vec.cend() - ( std::min((int)dls.fg_colors_vec.size(), (int) (view_colors::HI_COLOR_COUNT - 1))),
+                        dls.fg_colors_vec.cend(),
+                        [attrs](auto el) {return el == attrs.ta_fg_color;}
+                        ) ||
+                    std::any_of(
+                        dls.bg_colors_vec.cend() - ( std::min((int)dls.bg_colors_vec.size(), (int) (view_colors::HI_COLOR_COUNT - 1))),
+                        dls.bg_colors_vec.cend(),
+                        [attrs](auto el) {return el == attrs.ta_bg_color;}
+                    );
+                    suf += " "
+                ) {
+                    attrs = vc.attrs_for_ident(colname + suf);
+                    dls.dls_headers[lpc].hm_colorstr = colname + suf;
+                }
```

Comment:
I would prefer a method on `stacked_bar_chart` that checked if another ident had the same `text_attrs` value.

Reply:
Mmm I am not sure about a few things, so I will ask in case I misunderstood something. Apparently the colour assignment uses a modulo operation on `view_colors::HI_COLOR_COUNT` which equals `54`.
So I guess, correct me if I am wrong since I do not know the constraints there, that I can only have 54 colour combinations to choose from. Hence, if by any chance there are more than 54 columns needing colouring, unlikely as it may be, I would need to recycle those colours. In my code I've cheked that across the previous 53 columns the colour being chosen is absent. So checking whether another ident has the same `text_attrs` value would not solve the problem, since in the case of, let's say 70 numeric columns, more than one would have to use the same colours. Unless one associates a counter to each `text_attrs` colour combination in order to cycle through them once, twice, thrice, etcetera.
The other problem I see is the overlap with columns (and related colours) extracted from JSON structures, since one would need to process the non-JSON columns then all the JSON columns all over the records, but since there are only 54 combinations available it might just happen than for one JSON record two columns with the same assigned colours are adjacent to each other, while the same column might not be for a different JSON record.
I hope I've explained myself decently. Could you elaborate more on the value for `view_colors::HI_COLOR_COUNT` by any chance? I guess it is something terminal-related.

Reply:
The `view_colors::HI_COLOR_COUNT` number is the number of highlight colors that are usable from the 256-color xterm palette.   The `vc_highlight_colors` array is initialized here: 

https://github.com/tstack/lnav/blob/eaa6e50a523937401aab76f687b44c31cd788b3e/src/view_curses.cc#L460-L469

That loop tries to take samples of the different colors without picking colors right next to each other since they're hard to differentiate visually.  (That's what I mean by "usable".)

> So checking whether another ident has the same `text_attrs` value would not solve the problem

Not sure what you mean.  The `stacked_bar_chart` already has the attributes to be used for each identifier.  So, I'm not clear why a separate `fg_colors_vec` vector is needed.  It seems like it is accomplishing the same thing.

Reply:
Thanks @tstack . I had seen the code but had not attempted to understand what the algorithm meant. The choice of narrowing down the colours used to a more viable set for the purpose in question makes definite sense.
I will take another look at the code to understand your remark about `stacked_bar_chart` and ask you more details if comprehension evades me further.